### PR TITLE
Adjust zypper test for carbon

### PR DESCRIPTION
**Pull request by cachedout**
How does this look to you @meaksh ?

### CODE REVIEW

**Comment by meaksh** [comment]
@cachedout - Seems not only the unit tests need a fix but also the zypper module. I pushed https://github.com/meaksh/salt/commit/32f44b43ee027e297bda2c215d7d408d71a7ccdf to fix both module and tests.

**Comment by cachedout** [comment]
@meaksh. Great! Could you turn that into a PR against the carbon branch? I'll merge it right away. Thanks!

File: tests/unit/modules/zypper_test.py [hunk 1]
```diff
@@ -410,11 +412,12 @@ def __call__(self, *args, **kwargs):
...
             with patch('salt.modules.zypper.list_pkgs', MagicMock(side_effect=[{"vim": "1.1"}, {"vim": "1.1"}])):
-                ret = zypper.upgrade(dist_upgrade=True, fromrepo=["DUMMY"])
-                self.assertFalse(ret['result'])
-                self.assertEqual(ret['comment'], zypper_out.strip())
-                self.assertDictEqual(ret['changes'], {})
-                zypper_mock.noraise.call.assert_called_with('dist-upgrade', '--auto-agree-with-licenses', '--from', 'DUMMY')
+                try:
+                    self.assertRaises(CommandExecutionError, zypper.upgrade(dist_upgrade=True, fromrepo=["DUMMY"], dryrun=True))
+                except CommandExecutionError as exc:
+                    pass
```
Comment (isbm) [inline tests/unit/modules/zypper_test.py]: Always loving this :heart:

**Comment by meaksh** [comment]
@cachedout sure, on the way.. :smile:

**Comment by meaksh** [comment]
Carbon PR: https://github.com/saltstack/salt/pull/37220

**Pull request closed without merging**